### PR TITLE
Fix travis CI failure caused by git shallow clone

### PR DESCRIPTION
**Pull request by yufeih**
Travis is using shallow clone `git clone --depth=50 --branch=v3 https://github.com/dotnet/docfx.git dotnet/docfx`

https://travis-ci.org/dotnet/docfx/builds/380017122

### CODE REVIEW

File: src/docfx/lib/git/GitUtility.Native.cs [hunk 1]
```diff
@@ -95,8 +95,17 @@ private static unsafe List<Commit> LoadCommits(IntPtr repo)
...
-            while (NativeMethods.GitRevwalkNext(out var commitId, walk) == 0)
+            while (true)
             {
+                var error = NativeMethods.GitRevwalkNext(out var commitId, walk);
+
+                // https://github.com/libgit2/libgit2sharp/issues/1351
+                if (error == -3 /* GIT_ENOTFOUND */)
+                    throw new NotSupportedException($"Does not support git shallow clone");
+
+                if (error != 0)
```
Comment (OsmondJiang) [inline src/docfx/lib/git/GitUtility.Native.cs]: >error [](start = 20, length = 5)

just break?

File: test/docfx.Test/lib/GitUtilityTest.cs [hunk 2]
```diff
@@ -22,9 +23,17 @@ public static class GitUtilityTest
...
-            var lib = GitUtility.GetCommits(repo, new List<string> { pathToRepo })[0].ToList();
-            Assert.Equal(JsonConvert.SerializeObject(exe), JsonConvert.SerializeObject(lib));
+
+            try
+            {
+                var exe = await GitUtility.GetCommits(repo, pathToRepo);
+                var lib = GitUtility.GetCommits(repo, new List<string> { pathToRepo })[0].ToList();
+                Assert.Equal(JsonConvert.SerializeObject(exe), JsonConvert.SerializeObject(lib));
+            }
+            catch (NotSupportedException ex)
```
Comment (OsmondJiang) [inline test/docfx.Test/lib/GitUtilityTest.cs]: that means we never run this test successfully in CI? because CI is using shallow clone?

Comment (yufeih) [reply test/docfx.Test/lib/GitUtilityTest.cs]: only travis, on linux

Comment (OsmondJiang) [reply test/docfx.Test/lib/GitUtilityTest.cs]: git clone -b v3 --single-branch?

---
In reply to: [188843300](https://github.com/dotnet/docfx/pull/2781#discussion_r188843300) [](ancestors = 188843300)

Comment (yufeih) [reply test/docfx.Test/lib/GitUtilityTest.cs]: the git clone command is generated/managed by travis

**Pull request merged**